src/aof.c

```diff
@@ -2523,6 +2555,33 @@ void backgroundRewriteDoneHandler(int exitcode, int bysignal) {
         latencyEndMonitor(latency);
         latencyAddSampleIfNeeded("aof-rename", latency);
 
+        /* Rename the temporary incr aof file to 'new_incr_filename'. */
+        if (server.aof_state == AOF_WAIT_REWRITE) {
+            sds temp_incr_aof_name = sdscatprintf(sdsempty(), "%s%s%s", TEMP_FILE_NAME_PREFIX, server.aof_filename,
+                        INCR_FILE_SUFFIX);
+            sds temp_incr_filepath = makePath(server.aof_dirname, temp_incr_aof_name);
+            sds new_incr_filename = getNewIncrAofName(temp_am);
+            sds new_incr_filepath = makePath(server.aof_dirname, new_incr_filename);
+            latencyStartMonitor(latency);
+            if (rename(temp_incr_filepath, new_incr_filepath) == -1) {
+                serverLog(LL_WARNING,
+                    "Error trying to rename the temporary incr AOF file %s into %s: %s",
+                    temp_incr_aof_name,
+                    new_incr_filename,
+                    strerror(errno));
+                aofManifestFree(temp_am);
+                sdsfree(temp_incr_aof_name);
+                sdsfree(temp_incr_filepath);
+                sdsfree(new_incr_filepath);
+                goto cleanup;
```

Comment:
i think we need to also free new_base_filepath, and maybe also delete that file first (like what we do when failing to persist the manifest.
or maybe we need to move all that to the cleanup label, and sort out some error handling code in this function

Reply:
Agree, but currently all variables are defined inside the if branch, so we can't access them at cleanup

Reply:
i see you added a `sdsfree(new_base_filepath);`, but don't we also want to delete the file?

Reply:
You are right. I'm still struggling with what to do with that pile of `sdsfree`